internal/tenant/context.go

```
@@ -0,0 +1,39 @@
+package tenant
+
+import (
+	"context"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type contextKey int
+
+const tenantKey contextKey = iota
+
+// FromContext returns the tenant from a given context.
+func FromContext(ctx context.Context) *Tenant {
```

Should we return an error here if the tenant is not found? The caller might forget to check for the 0 tenant.

---

Hmm. Maybe? A little less convenient to use, but probably safer.
We could also return `(_ *Tenant, ok bool)`?

---

Either would work for me 👍. I know it's inconvenient but the little nudge might be worth it ;-)?

---

did that